### PR TITLE
Detect current numbering scheme on load

### LINKED ISSUE #67: Discover the existing heading preset on document load
Figure out what the current preset should be, based on the headings already in the document.

Alternately, we could add an extra invisible element that stores the preset, but that seems annoying to everything we've already done.

### PR DESCRIPTION
**Pull request by caffodian**
When the document content is replaced, look at the headings, and figure out which numbering scheme best fits the document.  

This is mainly for loading the doc, but also works for if someone replaces all the content via the source panel.  

fixes #67

### CODE REVIEW

**Comment by caffodian** [comment]
Insanely annoying bug - for some reason, dataReady fires before the html is set in one particular test, and never afterwards.

File: structuredheadings/plugin.js [hunk 1]
```diff
@@ -173,6 +173,12 @@
       ]
     };
 
+    editor.config.autonumberStylesWithClasses = Object.keys(
+      editor.config.autonumberStyles
+    ).filter(function (styleName) {
+      return editor.config.autonumberStyles[styleName] !== null;
+    });
```
Comment (caffodian) [inline structuredheadings/plugin.js:180]: this is for convenience later, mostly

File: structuredheadings/plugin.js [hunk 2]
```diff
@@ -232,6 +302,10 @@
       setupCommands(editor);
       setupStyles(editor);
 
+      editor.on("dataReady", function () {
+        self.currentScheme = self.detectScheme();
```
Comment (caffodian) [inline structuredheadings/plugin.js:306]: whenever the document data is replaced (setData or source window)

File: tests/structuredheadings/schemediscovery.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    },
+
+    "autodetect will detect a schemed document": function () {
+      var bot = this.editorBot;
+      var editor = bot.editor;
+
+      var listener = editor.on(
+        "dataReady",
+        function () {
+          listener.removeListener();
```
Comment (caffodian) [inline tests/structuredheadings/schemediscovery.js:141]: editor cleanup kinda sucks, so the listener has to remove itself (this is a pattern used in their tests, without which I would have been stuck on this a lot longer)

**Review by jlward — APPROVED** [review]

**Comment by jlward** [comment]
Code looks good. Reviewing JS is hard, but I don't see anything that jumps out as wrong.

**Pull request merged**